Added postgresql support to Crawler

Nibba2018 opened the pull request:
# Overview

* Removed **sqlite** support
* Added **postgresql** support
* Added .html to .gitignore
* Updated README with information for postgresql

Nibba2018 committed:
Removed sqlite support & Added postgresql support

All sqlite commands were replaced by postgresql commands.
Added *.html to gitignore as there were several html files generated during crawling.

Nibba2018 committed:
Updated README

Nibba2018 committed:
fixed a typo in README

sunitasen commented:
Hi! Nice work 😄 🎉 
One small suggestion, when you are working on an issue, first mention that in the issue. This avoids duplicate work.

sunitasen commented on crawler/vison/spiders/visonSpider.py:
```diff
@@ -1,16 +1,24 @@
-        conn = sqlite3.connect('vison.db')
-        c = conn.cursor()
+        
+        conn = psycopg2.connect(
```
Try handling the exception. What if the server is not connected?

Nibba2018 replied:
Sure will do so..

sunitasen commented on crawler/README.md:
```diff
@@ -1,26 +1,39 @@
 - Downloads
-1. Sqlite - <a href="https://www.sqlite.org/download.html">here</a> (make sure to add it to your path)
+
+    1.Sqlite - [here](https://www.postgresql.org/download/) (make sure to add it to your path)
```
Sqlite --> PostgreSQL

Nibba2018 replied:
Yeah I just made that change right now!!

Nibba2018 commented:
By mentioning the issue what did u exactly mean?

sunitasen commented:
@Nibba2018  which one?

Nibba2018 commented:
This one: ```One small suggestion, when you are working on an issue, first mention that in the issue. This avoids duplicate work.```

sunitasen commented:
You were working on issue #2 right?!
So before you start working on an issue, comment below the issue that you want to work on it. Then we can keep a track on who is working on which issue. Otherwise, there could be possibilities of conflicts, such as two people are working on the same issue. This will be a complete waste of time for one of them.

Nibba2018 committed:
Avoiding Crawler Infinite Loops & try/except block for database

Added changes to prevent the crawler from visiting the same URL again & again.
Included try/exception block regarding the interfacing of postgresql database.

Nibba2018 committed:
quick Bug Fix

Nibba2018 commented:
# Referencing issue #10 

## Changes:
* Added try/except block regarding the interfacing of postgreSQL database.
* Algorithm used to prevent revisiting URLS:
```
Consider two lists: 
 1. prevLinks
 2. nextLinks

> Store the list of all href links in nextLinks.
> If its the first url, then prevLinks list will be empty, hence we copy the contents of nextList and proceed with scrapping the URLS in nextList.
> Else prevLinks is not empty, we check whether the current URL is present in the list of nextLinks.
   > If present, then we remove the current url from the nextLinks list
   > else we remove URLs which are common to prevLinks and nextLinks from nextLinks
> Lastly, we check if nextList is empty or not
   > If empty then no further link is followed
   > else urls in nextLinks are crawled further.
```

sunitasen commented on crawler/vison/spiders/visonSpider.py:
```diff
@@ -3,43 +3,52 @@
-            host = "127.0.0.1",
-            database = "vison",
-            user = "postgres",
-            password = "9681"
```
Just a suggestion.. Don't include your own password! Leave it blank or just add "password"

Nibba2018 replied:
Actually the same credentials were provided in the previous version of the code, so I thought it would be better if I maintain the consistency. But anyway, I will make sure to take care of this in future.

sunitasen commented:
This looks good to me!
From next time, open a new pull request for new new issue(Don't forget to create a new branch from origin). This helps in keeping track of the changes made! 😄

Nibba2018 commented:
> 
> 
> This looks good to me!
> From next time, open a new pull request for new new issue(Don't forget to create a new branch from origin). This helps in keeping track of the changes made! 😄
Sure! will take care of it Next Time 👍
